java/src/main/java/ai/rapids/cudf/MemoryBuffer.java

```diff
@@ -149,7 +149,7 @@ final long getAddress() {
   /**
    * Close this buffer and free memory
    */
-  public void close() {
+  public synchronized void close() {
```

Comment:
Does this same synchronize-on-close need to happen on other classes with refcounting?  For example, I noticed `Scalar#incRefCount` was synchronized, curious why `Scalar#close` wasn't similarly updated as it is here.